routes/s3handler.js

```diff
@@ -0,0 +1,239 @@
+import express from 'express'
+import CryptoJS from 'crypto-js'
+import bodyParser from 'body-parser'
+import getObjectURLFromS3 from '../src/utils/s3'
+import faceUtils from '../src/utils/face++'
+
+const CLIENT_SECRET_KEY = process.env.AWS_SECRET_ACCESS_KEY
+const expectedMinSize = 0
+const expectedMaxSize = 5000000
+const EXPECTED_BUCKET = process.env.EXPECTED_BUCKET
+const EXPECTED_HOST_NAME = process.env.EXPECTED_HOST_NAME
+
+const router = express.Router()
+
+// Signs any requests.  Delegate to a more specific signer based on type of request.
+function signRequest(req, res) {
+    if (req.body.headers) {
+        signRestRequest(req, res);
+    }
+    else {
+        signPolicy(req, res);
+    }
+}
+
+// Signs multipart (chunked) requests.
+function signRestRequest(req, res) {
+    var version = req.query.v4 ? 4 : 2,
+        stringToSign = req.body.headers,
+        signature = version === 4 ? signV4RestRequest(stringToSign) : signV2RestRequest(stringToSign);
+
+    var jsonResponse = {
+        signature: signature
+    };
+
+    res.setHeader("Content-Type", "application/json");
+
+    if (isValidRestRequest(stringToSign, version)) {
+        res.end(JSON.stringify(jsonResponse));
+    }
+    else {
+        res.status(400);
+        res.end(JSON.stringify({invalid: true}));
```

Comment:
`res.status(400).send(...)`

Reply:
I think you can even use `res.json` instead of `res.send` then you can drop the stringify